### ISSUE TITLE
ZLUDA won't work with higher version of Blender 4.0+ (Blender 4.2 Alpha tested)

### ISSUE THREAD

**Issue by gioxyer** [vosen/ZLUDA#146]
When testing ZLUDA on Blender 4.2 Alpha it doesn't work, version tested Blender 4.2.0 [cfb60c98be7d], as main error display on viewport appear "Failed to configure CUDA context (Invalid context)".

Recognise correctly ZLUDA in render engine
![Screenshot (690)](https://github.com/vosen/ZLUDA/assets/161426313/9647d3ff-8c2b-4125-bfc3-f584931f2602)

Error "Failed to configure CUDA context (Invalid context)" in viewport
![Screenshot (691)](https://github.com/vosen/ZLUDA/assets/161426313/0bd16493-0a8d-4af2-b9ef-17bba4a67dc4)

Error "Failed to configure CUDA context (Invalid context)" in final render
![Screenshot (692)](https://github.com/vosen/ZLUDA/assets/161426313/76e027d2-d1d0-4216-b3ba-ab2060b476fd)

Display CMD main error
![Screenshot (693)](https://github.com/vosen/ZLUDA/assets/161426313/ad4e9007-4ace-4b3c-9b24-9a756e502e0e)

Main error:
Invalid context in cuStreamCreate(&cuda_stream_, CU_STREAM_NON_BLOCKING) (C:\Users\blender\git\blender-vdev\blender.git\intern\cycles\device\cuda\queue.cpp:21)
Invalid context in cuCtxPopCurrent(NULL) (C:\Users\blender\git\blender-vdev\blender.git\intern\cycles\device\cuda\util.cpp:19)
Invalid value in cuCtxPushCurrent(device->cuContext) (C:\Users\blender\git\blender-vdev\blender.git\intern\cycles\device\cuda\util.cpp:14)
Invalid context in cuCtxPopCurrent(NULL) (C:\Users\blender\git\blender-vdev\blender.git\intern\cycles\device\cuda\util.cpp:19)
Invalid context in cuDevicePrimaryCtxRelease(cuDevice) (C:\Users\blender\git\blender-vdev\blender.git\intern\cycles\device\cuda\device_impl.cpp:139)
Failed to configure CUDA context (Invalid context)

Refer to the Cycles GPU rendering documentation for possible solutions:
https://docs.blender.org/manual/en/latest/render/cycles/gpu_rendering.html

Invalid value in cuCtxPushCurrent(device->cuContext) (C:\Users\blender\git\blender-vdev\blender.git\intern\cycles\device\cuda\util.cpp:14)
Invalid context in cuStreamCreate(&cuda_stream_, CU_STREAM_NON_BLOCKING) (C:\Users\blender\git\blender-vdev\blender.git\intern\cycles\device\cuda\queue.cpp:21)
Invalid context in cuCtxPopCurrent(NULL) (C:\Users\blender\git\blender-vdev\blender.git\intern\cycles\device\cuda\util.cpp:19)
Invalid value in cuCtxPushCurrent(device->cuContext) (C:\Users\blender\git\blender-vdev\blender.git\intern\cycles\device\cuda\util.cpp:14)
Invalid context in cuCtxPopCurrent(NULL) (C:\Users\blender\git\blender-vdev\blender.git\intern\cycles\device\cuda\util.cpp:19)
Invalid context in cuDevicePrimaryCtxRelease(cuDevice) (C:\Users\blender\git\blender-vdev\blender.git\intern\cycles\device\cuda\device_impl.cpp:139)
Error: Failed to configure CUDA context (Invalid context)

**Comment by chocolatewafer** [comment]
some issues with the alpha build maybe? Blender 4.0 works fine for me with a RX 560X card.

**Comment by gioxyer** [comment]
With Blender 4.0 I didn't have any issue too, but if you try Blender 4.1 Release Candidate or Blender 4.2 Alpha seems that check if there is cuda cores inside Blender (cuCtxPushCurrent, cuStreamCreate, cuCtxPushCurrent, cuCtxPopCurrent)

**Comment by vosen** [comment]
Should be fixed as of the latest version

**Comment by gioxyer** [comment]
Hi, I tried to download the updated version, it show this main error, as version tested 35c6c11c7edf

![Screenshot (784)](https://github.com/vosen/ZLUDA/assets/161426313/5a8ef01a-b5d1-49f9-8cd6-7e4d4330063c)

Error: OsError { function: "DetourCreateProcessWithDllsW", error_code: 14001, message: "Unable to start the specified application. The application's side-by-side mode configuration is incorrect. Please consult the application event log or use the command-line tool sxstrace.exe to get more information." }

**Comment by vosen** [comment]
What happens if you start blender.exe normally? What happens if you check "Unblock" in blender.exe properties and try to run with ZLUDA?
Windows tends to block unknown freshly-download .exe and this might be the case here

**Comment by gioxyer** [comment]
You're right, seems windows blocking, I will try unzip again the file and then I will need proably to open blender.exe before zluda.

It says Unable to execute the App on your pc
![Screenshot 2024-03-28 185150](https://github.com/vosen/ZLUDA/assets/161426313/141d3d0d-eed8-4b27-82d7-8d8e77f9f9f8)

**Comment by gioxyer** [comment]
Now it open correctly, but seems give similar error, probably lower than before
![Screenshot (786)](https://github.com/vosen/ZLUDA/assets/161426313/20b6352e-1746-4cef-80f4-f94dd5fc83c7)

Invalid value in cuCtxPushCurrent(device->cuContext) (C:\Users\blender\git\blender-vdev\blender.git\intern\cycles\device\cuda\util.cpp:14)
Invalid context in cuStreamCreate(&cuda_stream_, CU_STREAM_NON_BLOCKING) (C:\Users\blender\git\blender-vdev\blender.git\intern\cycles\device\cuda\queue.cpp:21)
Invalid context in cuCtxPopCurrent(NULL) (C:\Users\blender\git\blender-vdev\blender.git\intern\cycles\device\cuda\util.cpp:19)

**Comment by vosen** [comment]
Did you build from source? If so what revision (shuld be 067c923408e962260bc4cfbf9374b3f75ca939ff)? The release binaries are for an older version.
I'm asking because I'm getting the same error when using previous git revision, but it works fine here with the newest one

**Comment by gioxyer** [comment]
I just use this version 067c923, but I add zluda.exe, because I didn't find in zluda folder with files used before, probably it could create conflict? 
![image](https://github.com/vosen/ZLUDA/assets/161426313/ba24f481-32b2-436a-99f3-4acf18feb1b0)

![Screenshot (789)](https://github.com/vosen/ZLUDA/assets/161426313/3ee7b491-4ba2-417f-a50e-217959290c91)

![Screenshot (790)](https://github.com/vosen/ZLUDA/assets/161426313/6d0909e9-4a36-4f48-a2ff-3fefdf48751b)

**Comment by vosen** [comment]
This won't work, one needs to build from source (and sorry, it should be 7d4147c8b2cc31422a73f5db3ec065db8af82246, though it's identical to 067c923).
If it's not urgent maybe wait a bit? I'll add an automatic build on each commit, so you will be able to download ready-to-use prerelease package. I did not foresee so much interest in the work-in-progress versions

**Comment by gioxyer** [comment]
No problem, I can safely wait. It was interesting to try some features of Blender 4.2, anyway in general zluda I must say is improving a lot, I also tried some fork versions of lshqqytiger, you are doing a great job. I had made a video about it and I must say there is a fair amount of interest in general, for the versions though Blender official and with other software always on 3d.

**Comment by BaalNetbek** [comment]
I compiled the latest version and it's working with 4.1. You can try yourself. [zluda-3-windows_7d4147c.zip](https://github.com/vosen/ZLUDA/files/14795187/zluda-3-windows_7d4147c.zip)

**Comment by gioxyer** [comment]
It works now with both version Blender 4.1 and Blender 4.2 Alpha, thank you so much for your help

![Screenshot (791)](https://github.com/vosen/ZLUDA/assets/161426313/a0b958d2-f6af-496e-a802-07ce900c5842)

![Screenshot (792)](https://github.com/vosen/ZLUDA/assets/161426313/265a20ab-53d6-4544-90e2-aa8da66f7c15)

**Comment by radu1006** [comment]
On windows with Blender 4.3 I get: The application was unable to start correctly (0xc0000142)
I used the latest release from 31 Dec 2024.

![Image](https://github.com/user-attachments/assets/26265001-29fa-4f0b-af71-af725f5cb1a1)

**Comment by arlekion** [comment]
> I compiled the latest version and it's working with 4.1. You can try yourself. [zluda-3-windows_7d4147c.zip](https://github.com/vosen/ZLUDA/files/14795187/zluda-3-windows_7d4147c.zip)

Will that work with 4.3, 4.4?

**Comment by BaalNetbek** [comment]
I may just add that old Zluda works for me with Blender 4.3.2.

**Comment by MI1255** [comment]
> I may just add that old Zluda works for me with Blender 4.3.2.

How were you able to launch?

**Comment by BaalNetbek** [comment]
Put this in cmd
PATH_TO_ZLUDA.EXE PATH_TO_BLENDER

**Comment by kode54** [comment]
Not working on Arch Linux, either with zluda stable or zluda-git AUR packages.
